app/react/PDF/components/PDF.js

```diff
@@ -189,16 +223,26 @@ PDF.defaultProps = {
   onPageChange: () => {},
   onPDFReady: () => {},
   style: {},
+  onTextSelection: () => {},
+  onTextDeselection: () => {},
+  references: Immutable.List(),
+  highlightReference: () => {},
+  activeReference: '',
 };
 
 PDF.propTypes = {
   onPageChange: PropTypes.func,
+  onTextSelection: PropTypes.func,
+  onTextDeselection: PropTypes.func,
   onPDFReady: PropTypes.func,
   file: PropTypes.string.isRequired,
   filename: PropTypes.string,
   onLoad: PropTypes.func.isRequired,
   pdfInfo: PropTypes.object,
   style: PropTypes.object,
+  references: PropTypes.instanceOf(Immutable.List),
```

Comment:
What alternative do you propose? I'm moving the render references to inside PDFPage as you suggested in a previous comment, so I could connect PDFPage to the store to get the references, but avoid connecting components is also a good idea, do you have another option?

Reply:
@konz when the references array change the only thing ideally that must re-render are the visibile page references.

the best alternative i can think of is to have a specific component that encapsulates the references per page.
this way you can still have the good semantics of passing down the references to every page as you do now. you still need to manage the rendering on PDFPage probably for the children to avoid the rendering of the highlights on not visible pages.

```<PDFPage><PageReferences page='1'></PDFPage>```